src/node_buffer.cc

```diff
@@ -424,12 +424,13 @@ MaybeLocal<Object> New(Environment* env,
                      True(env->isolate())).IsNothing()) {
     return Local<Object>();
   }
-  MaybeLocal<Uint8Array> ui = Buffer::New(env, ab, 0, length);
+  MaybeLocal<Uint8Array> maybe_ui = Buffer::New(env, ab, 0, length);
 
-  if (ui.IsEmpty())
+  Local<Uint8Array> ui;
+  if (!maybe_ui.ToLocal(&ui))
     return MaybeLocal<Object>();
 
-  return scope.Escape(ui.ToLocalChecked());
+  return scope.Escape(ui);
```

Comment:
is it important to call `scope.Escape` only if the handle is not empty ? Otherwise it looks like we could just `return Buffer::New(env, ab, 0, length)` ?

Reply:
I'll take a look and try that out, but it probably won't be until tomorrow though. Thanks

Reply:
There’s also `scope.EscapeMaybe()`, but the problem/the reason why this code looks a bit more complex than it should is that `MaybeLocal<>` doesn’t automatically cast up (`Uint8Array` → `Object`) unlike `Local<>`

Reply:
Okay, thanks for the explanation!

Reply:
> is it important to call scope.Escape only if the handle is not empty ?

I don't think there is any danger in doing so. My understanding is that if the value is empty the escape slot (which is a Handle in the enclosing HandleScope) in EscapableHandleScope will be set to the undefined value, and nullptr will be returned. 

I think that would cause a type mismatch between `MaybeLocal<Array>` and `MaybeLocal<Object>` if we pass the result from `Buffer::New` directly to `Escape`, but how about something like this:
```c++
return scope.Escape(New(env, ab, 0, length).FromMaybe(Local<Object>()));
```

Reply:
Sorry, I hadn't refreshed and did not see the other comments until after I commented.